src/compiler/parser/index.js

```diff
@@ -598,6 +598,16 @@ function makeAttrsMap (attrs: Array<Object>): Object {
     }
     map[attrs[i].name] = attrs[i].value
   }
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    map['v-model'] && (map['v-bind:value'] || map[':value']) &&
+    map['type'] !== 'checkbox' &&
+    map['type'] !== 'radio' &&
+    tag !== 'select'
+  ) {
+    var vBindValue = map['v-bind:value'] ? 'v-bind:value' : ':value'
+    warn(vBindValue + ' conflicts with v-model on the same element because the latter already expands to a value binding internally')
+  }
```

Comment:
Check looks good now - but I'm not sure if `makeAttrsMap` is the best place to do this check. Let's place it [here](https://github.com/vuejs/vue/blob/dev/src/platforms/web/compiler/directives/model.js#L25) since this warning is about `v-model`.

Reply:
Sure, it's now been moved